akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala

```diff
@@ -628,16 +628,19 @@ final class Broadcast[T](val outputPorts: Int, val eagerCancel: Boolean) extends
 }
 
 object Partition {
-
+  // FIXME make `PartitionOutOfBoundsException` a `final` class when possible
   case class PartitionOutOfBoundsException(msg: String) extends IndexOutOfBoundsException(msg) with NoStackTrace
 
   /**
-   * Create a new `Partition` stage with the specified input type.
+   * Create a new `Partition` stage with the specified input type. This method sets `eagerCancel` to `false`.
+   * To specify a different value for the `eagerCancel` parameter, then instantiate Partition using the constructor.
+   *
+   * If `eagerCancel` is true, partition cancels upstream if any of its downstreams cancel, if false, when all have cancelled.
    *
    * @param outputPorts number of output ports
    * @param partitioner function deciding which output each element will be targeted
-   */
-  def apply[T](outputPorts: Int, partitioner: T ⇒ Int): Partition[T] = new Partition(outputPorts, partitioner)
+   */ // FIXME BC add `eagerCancel: Boolean = false` parameter
```

Comment:
any problems with adding that param?

Reply:
Yeah, it's the overloaded method hell me and Viktor discussed... lemme find that

Reply:
Viktor's reproducer (shared with the scala team): 

```
Welcome to Scala 2.12.3 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_144).
Type in expressions for evaluation. Or try :help.

scala> object T {
     |   def apply[T](f: T => Int): Int = ???
     |   def apply[T](f: T => Int, dummy: Boolean): Int = ???
     | }
defined object T

scala> T[String](foo => 1)
<console>:13: error: missing parameter type
       T[String](foo => 1)
                 ^

scala> object T {
     |   def apply[T](f: T => Int): Int = ???
     |   //def apply[T](f: T => Int, dummy: Boolean): Int = ???
     | }
defined object T

scala> T[String](foo => 1)
scala.NotImplementedError: an implementation is missing
  at scala.Predef$.$qmark$qmark$qmark(Predef.scala:284)
  at T$.apply(<console>:12)
  ... 29 elided
```

Also known as: We're not able to add the overload because type inference blows up :(

Reply:
ok, then I suggest `.withEagerCancel(on: Boolean)` in `Partition` class

Reply:
ok, so `new` constructor overload works, interesting

Reply:
@patriknw Yes.

~~~scala
scala> final class Foo[T](f: T => Int, foo: Int) {
     |   def  this(f: T => Int) = this(f, 0)
     | }
defined class Foo

scala> val f = new Foo[String](s => s.toInt)
f: Foo[String] = Foo@42297bdf

scala> val f = new Foo[String](s => s.toInt, 1)
f: Foo[String] = Foo@e0d1dc4
~~~